### PR TITLE
Adding section on GNAT toolchain for embedded targets 

### PR DESCRIPTION
**Pull request by gusthoff**

### CODE REVIEW

File: content/courses/Ada_For_The_C_Embedded_Developer/chapters/02_Perspective.rst [hunk 1]
```diff
@@ -36,13 +36,67 @@ to GNAT, and sometimes to specific platforms supported by GNAT. We'll try to
...
+
+- **bareboard targets**, where the run-times do not depend on an operating
+  system.
+
+    - In this case, the application communicates directly with the device's
+      processor.
+
+For each platform, a set of run-time libraries is available. Run-time libraries
+implement a subset of the Ada language for different use cases, and they're
+different for each target platform. They may be selected via an attribute in
```
Comment (Robert-Tice) [inline content/courses/Ada_For_The_C_Embedded_Developer/chapters/02_Perspective.rst:72]: We should emphasize here that, although the run time libraries may vary from target to target, the user interface stays the same, providing portability for the application.

File: content/courses/Ada_For_The_C_Embedded_Developer/chapters/02_Perspective.rst [hunk 2]
```diff
@@ -36,13 +36,67 @@ to GNAT, and sometimes to specific platforms supported by GNAT. We'll try to
...
+    - They are known as Board Support Package |mdash| commonly refer as *BSP*.
+
+#. Code that is target-independent.
+
+    - This code implements language-defined functionality.
+
+The bareboard run-time libraries are provided as customized run-times that are
+configured to target a very specific micro-controller or processor. Therefore,
+for different micro-controllers and processors, the run-time libraries need to
+be ported to your target.
```
Comment (Robert-Tice) [inline content/courses/Ada_For_The_C_Embedded_Developer/chapters/02_Perspective.rst:92]: You can mention some examples here about what needs to be ported: Startup scripts, memory layouts, clock initializations, register descriptions, etc.

File: content/courses/Ada_For_The_C_Embedded_Developer/chapters/02_Perspective.rst (hunk 1 above)
Comment (gusthoff) [reply content/courses/Ada_For_The_C_Embedded_Developer/chapters/02_Perspective.rst:72]: I've just added this verbatim.

File: content/courses/Ada_For_The_C_Embedded_Developer/chapters/02_Perspective.rst [hunk 3]
```diff
@@ -44,10 +44,59 @@ course.
 The GNAT Toolchain for Embedded Targets
 -----------------------------------------
 
-.. todo::
+When we're discussing embedded programming, our target device is often
+different from the host, which is the device we're using to actually write and
+build an application. In this case, we're talking about compilation cross
```
Comment (pat-rogers) [inline content/courses/Ada_For_The_C_Embedded_Developer/chapters/02_Perspective.rst:49]: about cross compilation platforms.

**Pull request merged**